nilearn/tests/test_niimg_conversions.py

```diff
@@ -23,26 +23,30 @@
 from nilearn._utils.testing import assert_raises_regex
 
 
-class PhonyNiimage(object):
+class PhonyNiimage(nibabel.spatialimages.SpatialImage):
 
     def __init__(self):
         self.data = np.ones((9, 9, 9, 9))
-        self.affine = np.ones((4, 4))
+        self.my_affine = np.ones((4, 4))
 
     def get_data(self):
         return self.data
 
     def get_affine(self):
-        return self.affine
+        return self.my_affine
+
+    @property
```

Comment:
I guess so that you can use `bla.shape` rather than `bla.shape()`. Also IIRC properties are not settable by default so that trying `bla.shape = (3, 2, 1)` will fail.